### PR TITLE
add types for package jsreport

### PR DESCRIPTION
**Pull request by pofider**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## jsreport ([<kbd>unpkg</kbd>](https://unpkg.com/browse/jsreport@latest/))
was missing the following properties:
1. Reporter
1. renderDefaults
1. render
1. extendDefaults
1. reporter

<details>
<summary>as well as these 1 other properties...</summary>
<p>core</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6f7ad1a83675fe3b564654de02707569b699473d
</p>

File: types/jsreport/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Type definitions for jsreport 2.9
+// Project: https://github.com/jsreport/jsreport
+// Definitions by: pofider <https://github.com/jsreport>
```
Comment (elibarzilay) [inline types/jsreport/index.d.ts:3]: Is the URL a typo?

Even if it's an org that you belong to, it's better to have the URLs of the actual contributors here.

Comment (pofider) [reply types/jsreport/index.d.ts]: Yes, I fixed it with link to my profile https://github.com/pofider

File: types/jsreport/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+// import all availible types for jsreport included extensions
+import JsReportChromePdf = require('jsreport-chrome-pdf');
+import JsReportJsRender = require('jsreport-jsrender');
+import JReportHtmlToXlsx = require('jsreport-html-to-xlsx');
+import JsReportXlsx = require('jsreport-xlsx');
+
+// just reexport types from core
+import JsReport = require('jsreport-core');
+export = JsReport;
```
Comment (peterblazejewicz) [inline types/jsreport/index.d.ts:14]: @pofider 
I'm not sure if that is a proper way to reflect the module contrat, can you check this?
https://github.com/jsreport/jsreport/blob/master/index.js#L62-L75
(IMO module exports its own function and re-exports other types)

Comment (pofider) [reply types/jsreport/index.d.ts:14]: Thank you for reviewing. 
We were planning to remove the other exports in the next major, so I was thinking there is no point in adding types now.
I just wanted to re-export the types that are exported in [jsreport-core](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jsreport-core/index.d.ts) package. 
If it's an internal rule of the repository to cover all exports, I can add it temporarily for now.

Comment (peterblazejewicz) [reply types/jsreport/index.d.ts:14]: > If it's an internal rule of the repository to cover all exports, I can add it temporarily for now

Nope, only for a concern of providing same experience for users and reflecting the native module properly (in short, stick with the goal of the DT repo: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#L3).
I see the point. Also you own both implementations, so it's up to you (and for DT maintainer for final touch).

**Review by peterblazejewicz — APPROVED** [review]
@pofider thx!
LGTM!

**Comment by orta** [comment]
Cool, looks good for me too

**Pull request merged**